### ISSUE TITLE
spotlessIdeHook and configuration cache

### ISSUE THREAD

**Issue by nedtwigg** [diffplug/spotless#1082]
The spotless IDE hook is incompatible with the configuration cache, see https://github.com/ragurney/spotless-intellij-gradle/issues/18 for an example and stacktrace. Workaround for now is to specify `--no-configuration-cache` when using `spotlessIdeHook`.

**Comment by nedtwigg** [comment]
Initially, both the vscode and intelliJ plugins fixed this by adding `--no-configuration-cache` to their command lines:

- this is only needed if the user has enabled configuration cache with `org.gradle.unsafe.configuration-cache=true` in a `gradle.properties` somewhere
- if the user has not enabled that flag there is no effect

Unfortunately, for Gradle < 6.6, this flag is unrecognized and throws an exception
- https://github.com/badsyntax/vscode-spotless-gradle/issues/281

The VSCode plugin is trying to fix this by parsing and detecting the version, another way is to test it on first run, detect the error, then try again without that flag.

**Comment by lehmanju** [comment]
Would #1274 resolve this issue, if the configuration cache is natively supported and the current workaround with `JvmLocalCache` is removed?

We have several large projects which require a few minutes in the configuration phase. A fast IDE hook which makes use of the cache and doesn't invalidate it would be nice.

**Comment by nedtwigg** [comment]
Yes, #1274 will fix this.

**Comment by lehmanju** [comment]
With 7.0.0.BETA2 I get following error (running spotlessIdeHook with configuration cache enabled):
```
spotlessApply` of type `org.gradle.api.DefaultTask`: cannot serialize object of type 'com.diffplug.gradle.spotless.SpotlessTaskImpl', a subtype of 'org.gradle.api.Task', as these are not supported with the configuration cache.
```

**Comment by nedtwigg** [comment]
Should be fixed in `7.0.0.BETA4`, let me know if you have any issues.
